pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/TaskMetricsEmitter.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.controller.helix.core.minion;
+
+import java.util.Set;
+import org.apache.pinot.common.metrics.ControllerGauge;
+import org.apache.pinot.common.metrics.ControllerMetrics;
+import org.apache.pinot.controller.ControllerConf;
+import org.apache.pinot.controller.LeadControllerManager;
+import org.apache.pinot.controller.helix.core.minion.PinotHelixTaskResourceManager.TaskCount;
+import org.apache.pinot.core.periodictask.BasePeriodicTask;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class TaskMetricsEmitter extends BasePeriodicTask {
+  private static final Logger LOGGER = LoggerFactory.getLogger(TaskMetricsEmitter.class);
+  private final static String TASK_NAME = "TaskMetricsEmitter";
+
+  private final PinotHelixTaskResourceManager _helixTaskResourceManager;
+  private final ControllerMetrics _controllerMetrics;
+  private final LeadControllerManager _leadControllerManager;
+
+  public TaskMetricsEmitter(PinotHelixTaskResourceManager helixTaskResourceManager,
+      LeadControllerManager leadControllerManager, ControllerConf controllerConf, ControllerMetrics controllerMetrics) {
+    super(TASK_NAME, controllerConf.getTaskMetricsEmitterFrequencyInSeconds(),
+        controllerConf.getPeriodicTaskInitialDelayInSeconds());
+    _helixTaskResourceManager = helixTaskResourceManager;
+    _controllerMetrics = controllerMetrics;
+    _leadControllerManager = leadControllerManager;
+  }
+
+  @Override
+  protected final void runTask() {
+    // Make it so that only one controller returns the metric for all the tasks.
+    if (!_leadControllerManager.isLeaderForTable(TASK_NAME)) {
+      return;
+    }
+
+    // The call to get task types can take time if there are a lot of tasks.
```

Comment:
It'd be good to mention the default frequency is 5 mins here.

Reply:
I don't want to mention the number here -- if someone changes the default for some reason, they won't know to change the comment. I will add some comments to the effect that this is run fairly frequently.